src/main/java/com/google/devtools/build/lib/remote/AbstractRemoteActionCache.java

```diff
@@ -130,13 +131,19 @@ abstract void upload(
       return EMPTY_BYTES;
     }
     ByteArrayOutputStream bOut = new ByteArrayOutputStream((int) digest.getSizeBytes());
+    HashingOutputStream hashOut = digestUtil.newHashingOutputStream(bOut);
     SettableFuture<byte[]> outerF = SettableFuture.create();
     Futures.addCallback(
-        downloadBlob(digest, bOut),
+        downloadBlob(digest, hashOut),
```

Comment:
Doesn't this result in double hashing, since both concrete `downloadBlob` implementations now verify?